x-pack/plugins/threat_intelligence/public/modules/query_bar/components/filter_in/filter_in.tsx

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, VFC } from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiButtonEmpty, EuiButtonIcon, EuiContextMenuItem } from '@elastic/eui';
+import { Filter } from '@kbn/es-query';
+import { useIndicatorsFiltersContext } from '../../../indicators/hooks/use_indicators_filters_context';
+import { getIndicatorFieldAndValue } from '../../../indicators/lib/field_value';
+import { FilterIn as FilterInConst, updateFiltersArray } from '../../lib/filter';
+import { EMPTY_VALUE } from '../../../../../common/constants';
+import { Indicator } from '../../../../../common/types/indicator';
+import { useStyles } from './styles';
+
+export const ICON_TEST_ID = 'tiFilterInIcon';
+export const COMPONENT_TEST_ID = 'tiFilterInComponent';
+
+const ICON_TYPE = 'plusInCircle';
+const ICON_TITLE = i18n.translate('xpack.threatIntelligence.queryBar.filterInButton', {
+  defaultMessage: 'Filter In',
+});
+
+export interface FilterInProps {
```

Comment:
now that we have split components, there is _some_ code duplication, which isn't ideal but also not awful either... I could pull out things like this interface for example but I'm not entirely it's worth it, especially for code readability... thoughts?

Reply:
We should not sacrifice readability for deduping minor things, especially if this is about type defs imo. LGTM!